nbodykit/algorithms/paircount_tpcf/tpcf.py

```diff
@@ -188,6 +192,48 @@ def load(cls, output, comm=None):
         self.comm = comm
         return self
 
+    def to_xil(self, ells, mu_range=None):
+        r"""
+        Invert the measured wedges :math:`\xi(r,mu)` into correlation
+        multipoles, :math:`\xi_\ell(r)`.
+
+        Parameters
+        ----------
+        ells : array_like
+            the list of multipoles to compute
+        mu_range: array_like, optional
+            the range of :math:`\mu` to use to calculate multipoles (nearest '\mu' selected);
+            if not provided, all `\mu`-bins are used
+
+        Returns
+        -------
+        xil : BinnedStatistic
+            a data set holding the :math:`\xi_\ell(r)` multipoles
+        """
+        from scipy.special import legendre
+        from scipy.integrate import quad
+
+        # new data array
+        x = str(self.corr.dims[0])
+        dtype = numpy.dtype([(x, 'f8')] + [('corr_%d' %ell, 'f8') for ell in ells])
+        data = numpy.zeros((self.corr.shape[0]), dtype=dtype)
+        dims = [x]
+        edges = [self.corr.edges[x]]
+
+        if mu_range: sliced = self.corr.sel(mu=slice(*mu_range), method='nearest')
```

Comment:
Looks like mu_range is used only once here to perform a selection on the corr, and the only object the to_xil function operates on is self.corr. It sort of feels like the function may belong somewhere else. e.g. What if we make corr an instances of a subclass of BinnedStatistics, which adds a method to_xil (or to_poles) on top of existing methods of BinnedStatistics, that returns a new BinnedStatistics of poles?

The benefit is that the to_poles() method no longer needs to couple the selection logic.
```
     r = SurveyDataBox2PCF(.....)
     r.corr.sel(mu=slice(0.3, 0.5), method='nearest').to_poles(ells=[0, 2, 4])
```

I think this may require some changes in the BinnedStatistics class, allowing sel() to return a subclass instance.